libexec/pyenv

```diff
@@ -40,7 +40,9 @@ else
     local path="$1"
 
     while [ -n "$path" ]; do
-      cd "${path%/*}"
+      local parent="${path%/*}"
+      if [ "$path" = "$parent" ]; then break; fi
+      cd "$parent"
```

Comment:
Breaking is wrong here: the local file might be a symlink itself that needs to get resolved.